[DOCS] Update OpenVINO build instruction (Android system) (#24815)
openvinotoolkit/openvino: docs/dev/build_android.md

## Software requirements

- [CMake](https://cmake.org/download/) 3.13 or higher
- [SCons](https://scons.org/pages/download.html) 4.6.0 or higher
- [Android SDK Platform Tools](https://developer.android.com/tools/releases/platform-tools) (this guide has been validated with 30.0.0 release)
- [Android NDK](https://developer.android.com/ndk/downloads) (this guide has been validated with r26 release)

## How to build

> ### Create work directory and set environment variables for building

Reviewer: ```suggestion
### Create a work directory and set the environment variables for building
```

  ```sh
  mkdir openvino-android
  export OPV_HOME_DIR=${PWD}/openvino-android

>   export CURRENT_ANDROID_ABI=arm64-v8a

Reviewer 2: can we highlight that x86_64 can be used as well? E.g. that value to set depending on underlying HW

Author: @ilya-lavrenov updated, but below is also a description of which ABI can be used

  export CURRENT_ANDROID_PLATFORM=30
  export CURRENT_ANDROID_STL=c++_shared
  ```

>   * `ANDROID_ABI` specifies target architecture:
>      * `x86_64` for x64 build
>      * `armeabi-v7a with NEON` for ARM with NEON support
>      * `arm64-v8a` for ARM 64 bits
>   * `ANDROID_PLATFORM` - Android API version
>   * `ANDROID_STL` specifies that shared C++ runtime is used. 

Reviewer: ```suggestion
  * `ANDROID_ABI` specifies the target architecture:
     * `x86_64` for x64 build
     * `armeabi-v7a with NEON` for ARM with NEON support
     * `arm64-v8a` for ARM 64 bits
  * `ANDROID_PLATFORM` specifies the Android API version.
  * `ANDROID_STL` indicates that a shared C++ runtime is used. 
```

> ### Download and unpack Android packages  and 

Reviewer: ```suggestion
### Download and unpack Android packages 
```

* Download and unpack [Android NDK](https://developer.android.com/ndk/downloads)
```sh
  wget https://dl.google.com/android/repository/android-ndk-r26d-linux.zip --directory-prefix $OPV_HOME_DIR

  unzip $OPV_HOME_DIR/android-ndk-r26d-linux.zip -d $OPV_HOME_DIR
  mv $OPV_HOME_DIR/android-ndk-r26d $OPV_HOME_DIR/android-ndk
  export ANDROID_NDK_PATH=$OPV_HOME_DIR/android-ndk
```
* Download and unpack [Android SDK Platform Tools](https://developer.android.com/tools/releases/platform-tools)
```sh
  wget https://dl.google.com/android/repository/platform-tools-latest-linux.zip --directory-prefix $OPV_HOME_DIR

  unzip $OPV_HOME_DIR/platform-tools-latest-linux.zip -d $OPV_HOME_DIR
  export ANDROID_TOOLS_PATH=$OPV_HOME_DIR/platform-tools
  ```

> _For Windows and Mac operating systems, download and unpack steps are similar_

Reviewer: ```suggestion
_For Windows and Mac operating systems, the downloading and unpacking steps are similar._
```

### Build and install OneTBB™

> To improve parallelism performance of the OpenVINO™ library using OneTBB, it is required to separately build OneTBB for a specific version of andoird-ndk:

Reviewer: ```suggestion
To improve the parallelism performance of the OpenVINO™ library using OneTBB, it is required to separately build OneTBB for a specific version of the Android NDK:
```

  ```bash
  # Clone OneTBB™ repository 
  git clone --recursive https://github.com/oneapi-src/oneTBB $OPV_HOME_DIR/one-tbb
  # Create build and install directory 
  mkdir $OPV_HOME_DIR/one-tbb-build $OPV_HOME_DIR/one-tbb-install
  # Configure OneTBB™ CMake project 
  cmake -S $OPV_HOME_DIR/one-tbb \
        -B $OPV_HOME_DIR/one-tbb-build \
        -DCMAKE_BUILD_TYPE=Release \
        -DCMAKE_INSTALL_PREFIX=$OPV_HOME_DIR/one-tbb-install \

>         -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_PATH/build/cmake/android.toolchain.cmake \

Reviewer 2: can we set `CMAKE_TOOLCHAIN_FILE` globally? so, it can be used by both TBB and OpenVINO

Author: @ilya-lavrenov corrected

        -DANDROID_ABI=$CURRENT_ANDROID_ABI \
        -DANDROID_PLATFORM=$CURRENT_ANDROID_PLATFORM \
        -DANDROID_STL=$CURRENT_ANDROID_STL \
        -DTBB_TEST=OFF \
        -DCMAKE_SHARED_LINKER_FLAGS="-Wl,--undefined-version" 
  # Build OneTBB™ project 
  cmake --build $OPV_HOME_DIR/one-tbb-build --parallel
  # Install OneTBB™ project 
  cmake --install $OPV_HOME_DIR/one-tbb-build
  ```

### Build and install OpenVINO™
More information about building OpenVINO™ for android systems can be found [here](https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build_android.md).
  ```bash
  # Clone OpenVINO™ repository 
  git clone --recursive https://github.com/openvinotoolkit/openvino $OPV_HOME_DIR/openvino
  # Create build and install directory 
  mkdir $OPV_HOME_DIR/openvino-build $OPV_HOME_DIR/openvino-install
  # Configure OpenVINO™ CMake project 
  cmake -S $OPV_HOME_DIR/openvino \
        -B $OPV_HOME_DIR/openvino-build \

>         -DCMAKE_BUILD_TYPE=Release \

Reviewer 2: ```suggestion
```
because `Release` is used by default.

        -DCMAKE_INSTALL_PREFIX=$OPV_HOME_DIR/openvino-install \

>         -DENABLE_SAMPLES=ON \

Reviewer 2: it's `ON` default

Author: @ilya-lavrenov corrected

>         -DENABLE_TESTS=ON \

Reviewer 2: do we need to build tests? how they can be useful for users (not developers)?

        -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_PATH/build/cmake/android.toolchain.cmake \
        -DANDROID_ABI=$CURRENT_ANDROID_ABI \
        -DANDROID_PLATFORM=$CURRENT_ANDROID_PLATFORM \
        -DANDROID_STL=$CURRENT_ANDROID_STL \
        -DTBB_DIR=$OPV_HOME_DIR/one-tbb-install \

>         -DTBBROOT=$OPV_HOME_DIR/one-tbb-install

Reviewer 2: we can keep only `TBB_DIR` to point to folder with cmake files.

Author: @ilya-lavrenov corrected

  # Build OpenVINO™ project 
  cmake --build $OPV_HOME_DIR/openvino-build --parallel
  # Install OpenVINO™ project 
  cmake --install $OPV_HOME_DIR/openvino-build
  ```

> ### Download example model

Reviewer: ```suggestion
### Download the example model
```

  We will use `horizontal-text-detection-0001` example model
  ```bash
  mkdir $OPV_HOME_DIR/horizontal-text-detection-0001
  wget https://storage.openvinotoolkit.org/repositories/open_model_zoo/2022.3/models_bin/1/horizontal-text-detection-0001/FP32/horizontal-text-detection-0001.xml -P $OPV_HOME_DIR/horizontal-text-detection-0001/
  wget https://storage.openvinotoolkit.org/repositories/open_model_zoo/2022.3/models_bin/1/horizontal-text-detection-0001/FP32/horizontal-text-detection-0001.bin -P $OPV_HOME_DIR/horizontal-text-detection-0001/
  ```

> ### Use ADB tool to run example model and check library

Reviewer: ```suggestion
### Use the ADB tool to run the example model and check the library
```

  ```bash
  # Copy OpenVINO™ libraries and executables to android device

>   $ANDROID_TOOLS_PATH/adb push --sync $OPV_HOME_DIR/openvino/bin/aarch64/Release/* /data/local/tmp/

Reviewer 2: why not from installation folder?

Author: @ilya-lavrenov corrected
